File: frontend/src/api/model/models/rejectedAssociateInfoResponse.ts
/**
 * 거절된 준회원 정보 응답
 */
export interface RejectedAssociateInfoResponse {
  /** 사용자 고유 ID */
  userId?: number;
  /** 학번 */
  studentId?: string;
  /** 이름 */
  name?: string;
  /** 학과 */
  department?: string;
  /** 가입 동기 */
  motivation?: string;
  /** 가입 신청 일시 */
  createdAt?: string;
  /** 거절 사유 */
  rejectionReason?: string;
  /** 거절 일시 */
  rejectedAt?: string;
  /** 거절 처리자 ID */
  rejectedBy?: number;
}

File: backend/src/main/java/igrus/web/security/auth/approval/dto/response/RejectedAssociateInfoResponse.java
package igrus.web.security.auth.approval.dto.response;

import igrus.web.security.auth.approval.domain.AssociateDecision;
import igrus.web.user.domain.User;
import igrus.web.user.domain.Wish;

import java.time.Instant;
import java.util.List;

public record RejectedAssociateInfoResponse(
        Long userId,

        String studentId,

        String name,

        String department,

        String motivation,

        List<Wish> wishes,

        Instant createdAt,

        String rejectionReason,

        Instant rejectedAt,

        Long rejectedBy
) {
    public static RejectedAssociateInfoResponse from(AssociateDecision decision) {
        User user = decision.getUser();
        return new RejectedAssociateInfoResponse(
                user.getId(),
                user.getStudentId(),
                user.getName(),
                user.getDepartment(),
                user.getMotivation(),
                user.getWishes(),
                user.getCreatedAt(),
                decision.getReason(),
                decision.getDecidedAt(),
                decision.getDecidedBy()
        );
    }
}
